File: urbanairship-layout/src/main/java/com/urbanairship/android/layout/property/Direction.java
/* Copyright Airship and Contributors */

package com.urbanairship.android.layout.property;

import com.urbanairship.json.JsonException;

import java.util.Locale;

import androidx.annotation.NonNull;

public enum Direction {
    VERTICAL("vertical"),
    HORIZONTAL("horizontal");

    @NonNull
    private final String value;

    Direction(@NonNull String value) {
        this.value = value;
    }

    @NonNull
    public static Direction from(@NonNull String value) throws JsonException {
        for (Direction d : Direction.values()) {
            if (d.value.equals(value.toLowerCase(Locale.ROOT))) {
                return d;
            }
        }
        throw new JsonException("Unknown Direction value: " + value);
    }

    @NonNull
    @Override
    public String toString() {
        return name().toLowerCase(Locale.ROOT);
    }
}

File: urbanairship-layout/src/main/java/com/urbanairship/android/layout/property/Direction.kt
/* Copyright Airship and Contributors */
package com.urbanairship.android.layout.property

import com.urbanairship.json.JsonException
import com.urbanairship.json.JsonValue

public enum class Direction(
    private val value: String
) {
    VERTICAL("vertical"),
    HORIZONTAL("horizontal");

    override fun toString(): String {
        return name.lowercase()
    }

    public companion object {

        @Throws(JsonException::class)
        public fun from(value: JsonValue): Direction {
            val content = value.requireString().lowercase()

            return entries.firstOrNull { it.value == content }
                ?: throw JsonException("Unknown Direction value: $value")
        }
    }
}
